network/network.go

```diff
@@ -154,6 +140,13 @@ func NewNetworkInstance(
 		didDocumentFinder:   didDocumentFinder,
 		eventPublisher:      eventPublisher,
 		storeProvider:       storeProvider,
+		selfTestDialer: tls.Dialer{
+			NetDialer: &net.Dialer{
+				Timeout:   time.Second, // TODO: what is a reasonable timeout for this? It runs every couple of seconds when the /health endpoint is called
```

Comment:
I'd expect a typical TCP/TLS handshake to take 100-ish ms. So should be okay-ish, but we might have to adjust it later on.